_posts/2020-05-30-Metal3-Cluster-Deployment.md

```diff
@@ -0,0 +1,354 @@
+---
+title: "Metal³ development environment walkthrough part 2: Deploying a new baremetal cluster"
+draft: false
+categories: ["metal3", "kubernetes", "cluster API", "metal3-dev-env"]
+author: Himanshu Roy
+---
+
+## Introduction
+
+This blog post describes how to deploy a bare metal cluster, a virtual one for simplicity, using [Metal³](https://github.com/metal3-io/metal3-dev-env). We will briefly discuss the steps involved in setting up the cluster as well as some of the customizations available. If you want to know more about the architechture of Metal³, this [blogpost]({%post_url 2020-02-27-talk-kubernetes-finland-metal3 %}) can be helpful.
+
+This post builds upon the [**detailed metal3-dev-env walkthrough blogpost**]({%post_url 2020-02-18-metal3-dev-env-install-deep-dive %}) which describes in detail the steps involved in the environment set up and management cluster configuration. Here we will use that environment to deploy a new Kubernetes cluster using Metal³.
+
+Before we get started, there are a couple of requirements we are expecting to be fulfilled.
+
+<br/>
+<br/>
+
+## Requirements
+
+- Metal³ is already deployed and working,  if not please follow the instructions in the prevously mentioned - **detailed metal3-dev-env walkthrough blogpost**.
+- The appropriate environment variables are setup via shell or in the `config_${user}.sh` file, for example -
+  - CAPI_VERSION
+  - NUM_NODES
+  - CLUSTER_NAME
+
+<br/>
+<br/>
+
+## BareMetal Cluster Deployment
+
+The deployment scripts primarily use ansible and the existing Kubernetes management cluster (based on minikube ) for deploying the bare-metal cluster. Make sure that some of the environment varilables used for Metal³ deployment are set, if you didn't use `config_${user}.sh` for setting the environment varilables.
+
+| Parameter           | Description                  | Default                  |
+| ------------------- | ---------------------------- | ------------------------ |
+| CAPI_VERSION        | Version of Metal3 API4       | v1alpha3/v1alpha4        |
+| POD_CIDR            | Pod Network CIDR             | 192.168.0.0/18           |
+| CLUSTER_NAME        | Name of bare metal cluster   | test1                    |
+
+<br/>
+<br/>
+
+### Config Files and Resources Types
+
+!["A list of files under manifests directory for CLUSTER_NAME=eko"](/assets/2020-05-30-Metal3-Cluster-Deployment/manifest-directory.png)  
+
+<br/>
+
+Some of the configuration files generated as part of provisioning a cluster, in a centos based environment, are :
```

Comment:
How are these files generated? Who generates them, when, and where?

It's not clear from what has been mentioned so far in the blog post up until this point...

I think there should be a paragraph introducting the content of that section, giving an overview of the cluster provisioning process using the scripts, and explaining that a few configuration items and resource files will be created, that will be explained below.